custom_components/tahoma/alarm_control_panel.py

```diff
@@ -0,0 +1,173 @@
+        for device in data.get("entities").get(ALARM_CONTROL_PANEL)
+    ]
+    async_add_entities(entities)
+
+
+class TahomaAlarmControlPanel(TahomaDevice, AlarmControlPanelEntity):
+    """Representation of a TaHoma Alarm Control Panel."""
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        alarm_state = None
+
+        if self.has_state(MYFOX_ALARM_STATUS_STATE):
+            state = self.select_state(MYFOX_ALARM_STATUS_STATE)
+
+            if state == STATE_ARMED:
+                alarm_state = STATE_ALARM_ARMED_AWAY
+            elif state == STATE_DISARMED:
+                alarm_state = STATE_ALARM_DISARMED
+            elif state == STATE_PARTIAL:
+                alarm_state = STATE_ALARM_ARMED_NIGHT
+
+        if self.has_state(INTERNAL_CURRENT_ALARM_MODE_STATE):
+            state = self.select_state(INTERNAL_CURRENT_ALARM_MODE_STATE)
+
+            if state == STATE_OFF:
+                alarm_state = STATE_ALARM_DISARMED
+            elif state == STATE_ZONE_1:
+                alarm_state = STATE_ALARM_ARMED_HOME
+            elif state == STATE_ZONE_2:
+                alarm_state = STATE_ALARM_ARMED_NIGHT
+            elif state == STATE_TOTAL:
+                alarm_state = STATE_ALARM_ARMED_AWAY
+
+        if self.has_state(INTERNAL_CURRENT_ALARM_MODE_STATE) and self.has_state(
+            INTERNAL_TARGET_ALARM_MODE_STATE
+        ):
+            if self.select_state(
+                INTERNAL_CURRENT_ALARM_MODE_STATE
+            ) != self.select_state(INTERNAL_TARGET_ALARM_MODE_STATE):
+                alarm_state = STATE_ALARM_PENDING
+
+        if self.has_state(CORE_INTRUSION_STATE, INTERNAL_INTRUSION_DETECTED_STATE):
+            state = self.select_state(
+                CORE_INTRUSION_STATE, INTERNAL_INTRUSION_DETECTED_STATE
+            )
+
+            if state == STATE_DETECTED:
+                alarm_state = STATE_ALARM_TRIGGERED
+            elif state == STATE_PENDING:
+                alarm_state = STATE_ALARM_PENDING
+
+        return alarm_state
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        supported_features = 0
+
+        if self.has_command(COMMAND_ARM, COMMAND_ALARM_ON):
+            supported_features |= SUPPORT_ALARM_ARM_AWAY
+
+        if self.has_command(COMMAND_ALARM_PARTIAL_1):
+            supported_features |= SUPPORT_ALARM_ARM_HOME
+
+        if self.has_command(COMMAND_PARTIAL, COMMAND_ALARM_PARTIAL_2):
+            supported_features |= SUPPORT_ALARM_ARM_NIGHT
+
+        if self.has_command(COMMAND_SET_ALARM_STATUS):
+            supported_features |= SUPPORT_ALARM_TRIGGER
+            supported_features |= SUPPORT_ALARM_ARM_CUSTOM_BYPASS
+
+        return supported_features
+
+    async def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        await self.async_execute_command(
+            self.select_command(COMMAND_DISARM, COMMAND_ALARM_OFF)
```

Comment:
Why use select command, when you know the command?

Reply:
I am not sure where I copied this from.. Thanks for spotting this.

Reply:
@tetienne you need it to select one of them, cover does the same.